File: php/app/leetcode/s0001_two_sum/SolutionTest.php
<?php

use App\LeetCode\s0001_two_sum\Solution;
use Yosymfony\Toml\Toml;

$solution = new Solution();

function readTestFile(): array
{
    $cases = [];
    $test = Toml::parseFile(__DIR__ . "/../../../../test.toml");

    for ($i = 0; $i < count($test["table1"]["expected"]); $i++) {
        $cases["示例" . ($i + 1)] = [
            "expected" => $test["table1"]["expected"][$i],
            "nums" => $test["table1"]["nums"][$i],
            "target" => $test["table1"]["target"][$i],
        ];
    }

    return $cases;
}

test("1.两数之和", function (array $expected, array $nums, int $target) use ($solution): void {
    expect($solution->twoSum($nums, $target))->toEqual($expected);
})->with(readTestFile());

File: golang/src/leetcode/s0001_two_sum/solution_test.go
package s0001_two_sum

import (
	"github.com/BurntSushi/toml"
	"github.com/stretchr/testify/assert"
	"log"
	"testing"
)

type Test struct {
	Table1 Owner `toml:"table1"`
}

type Owner struct {
	Expected [][]int `toml:"expected"`
	Nums     [][]int `toml:"nums"`
	Target   []int   `toml:"target"`
}

type Case struct {
	expected []int
	nums     []int
	target   int
}

var cases []Case

func readTestFile() {
	var test Test

	if _, err := toml.DecodeFile("../../../../test.toml", &test); err != nil {
		log.Fatal(err)
	}

	for i := 0; i < len(test.Table1.Expected); i++ {
		cases = append(cases, Case{
			expected: test.Table1.Expected[i],
			nums:     test.Table1.Nums[i],
			target:   test.Table1.Target[i]},
		)
	}
}

func TestTwoSum(t *testing.T) {
	readTestFile()

	for _, c := range cases {
		t.Run("1.两数之和", func(t *testing.T) {
			assert.Equal(t, c.expected, twoSum(c.nums, c.target))
		})
	}
}
